pkg/ssh/ssh.go

```diff
@@ -291,6 +292,33 @@ func ParsePublicKeyFromFile(keyFile string) (*rsa.PublicKey, error) {
 	return rsaKey, nil
 }
 
+func MakePrivateKeySignerFromAgent(keyComment string) (ssh.Signer, error) {
+	sockName := os.Getenv("SSH_AUTH_SOCK")
+	if len(sockName) == 0 {
+		return nil, fmt.Errorf("SSH_AUTH_SOCK not defined in the environment")
+	}
+	socket, err := net.DialTimeout("unix", sockName, time.Minute)
+	if err != nil {
+		return nil, fmt.Errorf("could not connect to agent: %v", err)
+	}
+	signers, err := agent.NewClient(socket).Signers()
```

Comment:
Now that you have a successful socket, do you need to `defer socket.Close()`?

Reply:
I think the agent client takes ownership of the socket. It needs it to talk to the agent.

Reply:
Yeah, I wasn't sure how long it needed to stay open. Can you confirm?

Reply:
Yes, the agent client keeps it around for later use: https://github.com/golang/crypto/blob/master/ssh/agent/client.go#L264
It's the agent who has the secret, so this needs to stay alive as long as we keep making ssh connections.

Reply:
Cool, thanks for digging. I think we probably want a way to close the connection when we're done with it? cc @liggitt

Reply:
Looks like achieving this would require some changes to the users of `GetSigner`?
Alternatively we could cache and reuse the agent client, to ensure we don't leak an unbounded number of sockets. WDYT?

Reply:
@ncdc @liggitt can I get an answer on this before investing time into rebasing this PR?

Reply:
I'm not super familiar with where this code is used, so I'm not sure of the best approach. Leaking anything, however, never sounds like a good thing, but I'm not sure we can easily avoid it?

Reply:
Can you add a TODO about the leaked socket? Or would this be an appropriate usage of runtime.SetFinalizer? I don't want that to disappear.

Reply:
I think I was looking for something like this, I wonder how I missed `SetFinalizer`. After reading https://groups.google.com/forum/#!topic/golang-dev/DMiUkpS1uyQ it seems to me like this would indeed be an appropriate use of it here:
> useful only for releasing non-memory resources associated with an object during a long-running program